apps/dashboard/src/@3rdweb-sdk/react/hooks/useContractRoles.ts

```diff
@@ -91,20 +92,20 @@ export function useIsMinter<TContract extends ValidContractInstance>(
   return isAccountRole;
 }
 
-export function useIsLister<TContract extends ValidContractInstance>(
-  contract: RequiredParam<TContract>,
-) {
+// Applicable to Marketplace v3 only
+export function useIsLister(contract: ThirdwebContract) {
   const address = useActiveAccount()?.address;
-  const { data: contractType } = useContractType(contract?.getAddress());
-  const contractHasRoles = isContractWithRoles(contract);
-  const isAccountRole = useIsAccountRole(
-    "lister",
-    contractHasRoles ? contract : undefined,
-    address,
-  );
+  const { data: userCanList } = useReadContract(hasRole, {
```

Comment:
can we add an enabled check on this (queryOptions) to only run if we have an address? otherwise we'll be making rpc calls even if no user is connected

Reply:
on it

Reply:
done